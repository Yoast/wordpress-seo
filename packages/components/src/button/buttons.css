.yoast .yoast-button {
	display: inline-flex;
	align-items: center;
	justify-content: center;
	position: relative;
	padding: 10px 12px 12px;
	text-decoration: none;
	border-radius: 4px;
	border: 1px solid rgba(0, 0, 0, 0.2);
	box-shadow: inset 0 -2px 0 rgba(0, 0, 0, 0.1);
	transition: background-color 150ms ease-out 0s;
	line-height: 1.2;
	font-size: 14px;
	cursor: pointer;
}

.yoast .yoast-button:focus,
.yoast-close:focus,
.yoast-remove:focus,
.yoast-hide:focus {
	box-shadow: var(--yoast-color-focus);
	outline: none;
}

.yoast .yoast-button::-moz-focus-inner,
.yoast-close::-moz-focus-inner,
.yoast-remove::-moz-focus-inner,
.yoast-hide::-moz-focus-inner {
	border: 0;
}

.yoast .yoast-button:not(:disabled):active { /* 'Pressed' button state */
	box-shadow: none;
	top: 2px;
}

.yoast .yoast-button:disabled {
	opacity: 0.5;
	cursor: default;
}

.yoast .yoast-button--primary {
	color: var(--yoast-color-white);
	background-color: var(--yoast-color-primary);
	border: 1px solid rgba(0, 0, 0, 0.2);
}

.yoast .yoast-button--primary:visited {
	color: var(--yoast-color-white);
	border: 1px solid rgba(0, 0, 0, 0.2);
}

.yoast .yoast-button--primary:not(:disabled):hover,
.yoast .yoast-button--primary:active {
	color: var(--yoast-color-white);
	background-color: var(--yoast-color-primary-darker);
	border: 1px solid rgba(0, 0, 0, 0.2);
}

.yoast .yoast-button--primary:focus {
	color: var(--yoast-color-white);
	background-color: var(--yoast-color-primary);
}

.yoast .yoast-button--secondary {
	color: var(--yoast-color-dark);
	background-color: var(--yoast-color-secondary);
	box-shadow: inset 0 -2px 0 rgba(0, 0, 0, 0.1);
}

.yoast .yoast-button--secondary:visited {
	color: var(--yoast-color-dark);
}

.yoast .yoast-button--secondary:not(:disabled):hover,
.yoast .yoast-button--secondary:active {
	color: var(--yoast-color-dark);
	background-color: var(--yoast-color-secondary-darker);
	border: 1px solid rgba(0, 0, 0, 0.2);
}

.yoast .yoast-button--buy {
	color: var(--yoast-color-dark);
	background-color: var(--yoast-color-sale);
}

.yoast .yoast-button--buy:visited {
	color: var(--yoast-color-dark);
}

.yoast .yoast-button--buy:not(:disabled):hover,
.yoast .yoast-button--buy:active {
	color: var(--yoast-color-dark);
	background-color: var(--yoast-color-sale-darker)
}

.yoast .yoast-button--buy__caret {
	flex-shrink: 0;
	display: inline-block;
	width: 6px;
	height: 16px;
	/*Negative margin to compensate the icon white space within its viewbox.*/
	margin: 0 -2px 0 6px;
	background-color: currentColor;
	-webkit-mask-image: var(--yoast-svg-icon-caret-right);
	mask-image: var(--yoast-svg-icon-caret-right);
}

.yoast .yoast-button--edit {
	width: 20.25px;
	height: 18px;
	flex-shrink: 0;
	display: inline-block;
	background-color: currentColor;
	-webkit-mask-image: var(--yoast-svg-icon-edit);
	mask-image: var(--yoast-svg-icon-edit);
	margin-right: 8px;
}

html[dir=rtl] .yoast .yoast-button--edit {
	margin-right: 0;
	margin-left: 8px;
}

html[dir=rtl] .yoast .yoast-button--buy {
	flex-direction: row-reverse;
}

.yoast .yoast-button--small {
	font-size: 13px;
	padding: 5px 8px 8px;
}

.yoast .yoast-button--small .yoast-button--buy__caret {
	width: 4px;
	height: 10px;
}

.yoast-remove, .yoast-hide {
	color: #dc3232;
	background-color: transparent;
	border: none;
	cursor: pointer;
	text-decoration: underline;
	padding: 0;
	font-size: 14px;
}

.yoast-hide {
	color: var(--yoast-color-link);
}

.yoast-field-group__upload .yoast-button {
	margin-right: 24px;
}

/* Close button for modals or papers */
.yoast-close {
	display: flex;
	justify-content: center;
	align-items: center;
	border: none;
	background: none;
	height: 44px;
	width: 44px;
	cursor: pointer;
	box-shadow: none;
	padding: 0;
}

.yoast-close svg {
	fill: var(--yoast-color-default);
	width: 14px;
}

@media screen and (max-width: 782px) {
	.yoast-close svg {
		width: 10px;
	}
}
