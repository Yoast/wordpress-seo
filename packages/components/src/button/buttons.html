<!--A button can both be used for a link and a button.
Use a link when the button should redirect to another page
use a button when it's used for an action: for example, opening a modal.-->

<!DOCTYPE html>
<html lang="en">

<head>

	<meta charset="utf-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">

	<style>
		/* Demo styling */
		body {
			font-family: var(--yoast-font-family);
			font-size: var(--yoast-font-size-default);
			color: var(--yoast-color-font-default);
			line-height: 1.5;
		}
	</style>

	<link rel='stylesheet' href='../base/colors.css' type='text/css' media='all' />
	<link rel='stylesheet' href='../base/typography.css' type='text/css' media='all' />
	<link rel='stylesheet' href='../base/borders.css' type='text/css' media='all' />
	<link rel='stylesheet' href='../base/utility.css' type='text/css' media='all' />
	<link rel='stylesheet' href='../base/icons.css' type='text/css' media='all' />

	<link rel='stylesheet' href='buttons.css' type='text/css' media='all' />

</head>

<body class="yoast">
	<p>
		You can use both a <code>button</code> and a <code><a href=""></a></code> with the buttons.
		A button is used for an action, an a tag for a link to (external) content.
	</p>

	<p>Primary, purple button. Use with class <code>yoast-button yoast-button--primary</code></p>

	<button class="yoast-button yoast-button--primary" type="button">Save changes</button>

	<p>Secondary, gray button. Use with class <code>yoast-button yoast-button--secondary</code></p>

	<button class="yoast-button yoast-button--secondary" type="button">Save changes</button>

	<p>Disabled secondary, gray button. Use with class <code>yoast-button yoast-button--secondary</code></p>

	<button disabled class="yoast-button yoast-button--secondary" type="button">Save changes</button>

	<p>Upsell, yellow button. Use with class <code>yoast-button yoast-button--buy</code></p>

	<a href="#" class="yoast-button yoast-button--buy">Buy premium</a>

	<p>Upsell, yellow button with arrow. Use with class <code>yoast-button yoast-button--buy</code> and in the span use
		<code>yoast-button--buy__caret</code></p>

	<a href="#" class="yoast-button yoast-button--buy">Buy premium<span class="yoast-button--buy__caret"></span></a>

	<p>Small upsell, yellow button with arrow. Use with class
		<code>yoast-button yoast-button--buy yoast-button--small</code> and in the span use
		<code>yoast-button--buy__caret</code></p>

	<a href="#" class="yoast-button yoast-button--buy yoast-button--small">Buy premium<span
			class="yoast-button--buy__caret"></span></a>

	<p>Remove button. Use with class <code>yoast-remove</code></p>

	<input class="yoast-remove" value="Remove button" type="button" />

	<p>Hide button. Use with class <code>yoast-hide</code></p>

	<button class="yoast-hide" type="button">Hide button</button>

	<p>Close button</p>

	<button class="yoast-close" aria-label="Close" type="button">
		<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 352 512" role="img" aria-hidden="true" focusable="false">
			<path
				d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z">
			</path>
		</svg>
	</button>

</body>

</html>
