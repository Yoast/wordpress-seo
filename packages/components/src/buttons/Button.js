// External dependencies.
import styled from "styled-components";
import flow from "lodash/flow";
import PropTypes from "prop-types";

// Yoast dependencies.
import { colors, rgba } from "@yoast/style-guide";
import { getDirectionalStyle } from "@yoast/helpers";

const settings = {
	minHeight: 32,
	verticalPadding: 4,
	borderWidth: 1,
};

const ieMinHeight = settings.minHeight - ( settings.verticalPadding * 2 ) - ( settings.borderWidth * 2 );

/**
 * Returns a component with applied base button styles.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied base button styles.
 */
export function addBaseStyle( component ) {
	return styled( component )`
		display: inline-flex;
		align-items: center;
		justify-content: center;
		vertical-align: middle;
		border-width: ${ `${ settings.borderWidth }px` };
		border-style: solid;
		margin: 0;
		padding: ${ `${ settings.verticalPadding }px` } 10px;
		border-radius: 3px;
		cursor: pointer;
		box-sizing: border-box;
		font-size: inherit;
		font-family: inherit;
		font-weight: inherit;
		text-align: ${ getDirectionalStyle( "left", "right" ) };
		overflow: visible;
		min-height: ${ `${ settings.minHeight }px` };
		transition: var(--yoast-transition-default);

		svg {
			// Safari 10
			align-self: center;
		}

		// Only needed for IE 10+. Don't add spaces within brackets for this to work.
		@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
			::after {
				display: inline-block;
				content: "";
				min-height: ${ `${ ieMinHeight }px` };
			}
		}
	`;
}

/**
 * Returns a component with applied focus styles.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied focus styles.
 */
export function addFocusStyle( component ) {
	return styled( component )`
		&::-moz-focus-inner {
			border-width: 0;
		}

		&:focus {
			outline: none;
			border-color: ${ props => props.focusBorderColor };
			color: ${ props => props.focusColor };
			background-color: ${ props => props.focusBackgroundColor };
			box-shadow: 0 0 3px ${ props => rgba( props.focusBoxShadowColor, .8 ) }
		}
	`;
}

/**
 * Returns a component with applied hover styles.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied hover styles.
 */
export function addHoverStyle( component ) {
	return styled( component )`
		&:hover {
			color: ${ props => props.hoverColor };
			background-color: ${ props => props.hoverBackgroundColor };
			border-color: var(--yoast-color-border--default);
		}
	`;
}

/**
 * Returns a component with applied active styles.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied active styles.
 */
export function addActiveStyle( component ) {
	return styled( component )`
		&:active {
			color: ${ props => props.activeColor };
			background-color: ${ props => props.activeBackgroundColor };
			border-color: ${ props => props.hoverBorderColor };
			box-shadow: inset 0 2px 5px -3px ${ props => rgba( props.activeBorderColor, 0.5 ) }
		}
	`;
}

/**
 * Returns a component with all button selector styles applied.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied styles.
 */
export const addButtonStyles = flow( [
	/*
	 * Styled-components applies the generated CSS classes in a reversed order,
	 * but we want them in the order: base - hover - focus - active.
	 */
	addActiveStyle,
	addFocusStyle,
	addHoverStyle,
	addBaseStyle,
] );

/**
 * Returns a basic styled button.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} styled button.
 */
export const BaseButton = addButtonStyles(
	styled.button`
		color: ${ props => props.textColor };
		border-color: ${ props => props.borderColor };
		background: ${ props => props.backgroundColor };
		box-shadow: 0 1px 0 ${ props => rgba( props.boxShadowColor, 1 ) };
	`
);

BaseButton.propTypes = {
	type: PropTypes.string,
	backgroundColor: PropTypes.string,
	textColor: PropTypes.string,
	borderColor: PropTypes.string,
	boxShadowColor: PropTypes.string,
	hoverColor: PropTypes.string,
	hoverBackgroundColor: PropTypes.string,
	activeColor: PropTypes.string,
	activeBackgroundColor: PropTypes.string,
	activeBorderColor: PropTypes.string,
	focusColor: PropTypes.string,
	focusBackgroundColor: PropTypes.string,
	focusBorderColor: PropTypes.string,
	focusBoxShadowColor: PropTypes.string,
};

BaseButton.defaultProps = {
	type: "button",
	backgroundColor: colors.$color_button,
	textColor: colors.$color_button_text,
	borderColor: colors.$color_button_border,
	boxShadowColor: colors.$color_button_border,
	hoverColor: colors.$color_button_text_hover,
	hoverBackgroundColor: colors.$color_button_hover,
	activeColor: colors.$color_button_text_hover,
	activeBackgroundColor: colors.$color_button,
	activeBorderColor: colors.$color_button_border_active,
	focusColor: colors.$color_button_text_hover,
	focusBackgroundColor: colors.$color_white,
	focusBorderColor: colors.$color_blue,
	focusBoxShadowColor: colors.$color_blue_dark,
};

/**
 * Returns a styled Button with set font size.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} styled button.
 */
export default BaseButton;
