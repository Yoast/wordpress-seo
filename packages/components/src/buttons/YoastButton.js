// External dependencies.
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

// Yoast dependencies.
import { colors, rgba } from "@yoast/style-guide";

const settings = {
	minHeight: 48,
	verticalPadding: 0,
	borderWidth: 0,
};

const ieMinHeight = settings.minHeight - ( settings.verticalPadding * 2 ) - ( settings.borderWidth * 2 );

/**
 * Returns a component with a Yoast button-like style.
 *
 * The styles provided here are meant to be applied to both button and link
 * elements that need to look like styled buttons.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied button styles.
 */
export function addButtonStyles( component ) {
	return styled( component )`
		display: inline-flex;
		align-items: center;
		justify-content: center;
		vertical-align: middle;
		min-height: ${ `${ settings.minHeight }px` };
		margin: 0;
		padding: 0 16px;
		padding: ${ `${ settings.verticalPadding }px` } 16px;
		border: 0;
		border-radius: 4px;
		box-sizing: border-box;
		font: 400 14px/24px "Open Sans", sans-serif;
		text-transform: uppercase;
		box-shadow: 0 2px 8px 0 ${ rgba( colors.$color_black, 0.3 ) };
		transition: box-shadow 150ms ease-out;

		&:hover,
		&:focus,
		&:active {
			box-shadow:
				0 4px 10px 0 ${ rgba( colors.$color_black, 0.2 ) },
				inset 0 0 0 100px ${ rgba( colors.$color_black, 0.1 ) };
			color: ${ props => props.textColor };
		}

		&:active {
			transform: translateY( 1px );
			box-shadow: none;
		}

		// Only needed for IE 10+. Don't add spaces within brackets for this to work.
		@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
			::after {
				display: inline-block;
				content: "";
				min-height: ${ `${ ieMinHeight }px` };
			}
		}
	`;
}

/**
 * Returns a component with a button and an inner span element.
 *
 * The inner span is only needed to fix a Safari 10 bug with flexbox and button
 * elements. This bug is fixed in Safari Technology Previs and in the future it
 * will be possible to remove this component and directly style a button element.
 * See https://github.com/philipwalton/flexbugs#9-some-html-elements-cant-be-flex-containers
 *
 * @returns {ReactElement} The button with inner span.
 */
export const YoastButtonBase = ( { className, onClick, type, children, isExpanded } ) => (
	<button className={ className } onClick={ onClick } type={ type } aria-expanded={ isExpanded }>
		<span>
			{ children }
		</span>
	</button>
);

YoastButtonBase.propTypes = {
	className: PropTypes.string,
	onClick: PropTypes.func,
	type: PropTypes.string,
	isExpanded: PropTypes.bool,
	children: PropTypes.oneOfType( [
		PropTypes.arrayOf( PropTypes.node ),
		PropTypes.node,
		PropTypes.string,
	] ),
};

YoastButtonBase.defaultProps = {
	type: "button",
};

/**
 * Returns a Button with the Yoast button style.
 *
 * See the Safari 10 bug description in the YoastButtonBase JSDoc.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} styled button.
 */
export const YoastButton = addButtonStyles(
	styled( YoastButtonBase )`
		color: ${ props => props.textColor };
		background: ${ props => props.backgroundColor };
		min-width: 152px;
		${ props => props.withTextShadow ? `text-shadow: 0 0 2px ${ colors.$color_black }` : "" };
		overflow: visible;
		cursor: pointer;

		&::-moz-focus-inner {
			border-width: 0;
		}

		// Only needed for Safari 10 and only for buttons.
		span {
			display: inherit;
			align-items: inherit;
			justify-content: inherit;
			width: 100%;
		}
	`
);

YoastButton.propTypes = {
	backgroundColor: PropTypes.string,
	textColor: PropTypes.string,
	withTextShadow: PropTypes.bool,
};

YoastButton.defaultProps = {
	backgroundColor: colors.$color_green_medium_light,
	textColor: colors.$color_white,
	withTextShadow: true,
};
