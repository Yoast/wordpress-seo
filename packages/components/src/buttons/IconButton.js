// External dependencies.
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import omit from "lodash/omit";

// Yoast dependencies.
import { getDirectionalStyle } from "@yoast/helpers";

// Internal dependencies.
import { default as Button } from "./Button";
import { SvgIcon } from "../index";

/**
 * Applies styles to SvgIcon for IconButton with text.
 *
 * @param {ReactElement} icon The original SvgIcon.
 *
 * @returns {ReactElement} SvgIcon with text styles applied.
 */
function addIconTextStyle( icon ) {
	return styled( icon )`
		margin: ${ getDirectionalStyle( "0 8px 0 0", "0 0 0 8px" ) };
		flex-shrink: 0;
	`;
}

/**
 * Returns an icon button that can optionally contain text.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} styled icon button.
 */
const IconButton = ( props ) => {
	const { children: text, icon, iconColor } = props;

	let IconComponent = SvgIcon;
	if ( text ) {
		IconComponent = addIconTextStyle( IconComponent );
	}

	const newProps = omit( props, "icon" );

	return (
		<Button { ...newProps }>
			<IconComponent icon={ icon } color={ iconColor } />
			{ text }
		</Button>
	);
};

IconButton.propTypes = {
	icon: PropTypes.string.isRequired,
	iconColor: PropTypes.string,
	children: PropTypes.oneOfType( [
		PropTypes.arrayOf( PropTypes.node ),
		PropTypes.node,
		PropTypes.string,
	] ),
};

IconButton.defaultProps = {
	iconColor: "#000",
};

export default IconButton;
