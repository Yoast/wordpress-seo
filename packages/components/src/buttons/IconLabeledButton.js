// External dependencies.
import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import flow from "lodash/flow";
import omit from "lodash/omit";

// Yoast dependencies.
import { colors } from "@yoast/style-guide";
import SvgIcon from "../SvgIcon";

// Internal dependencies.
import { addActiveStyle, addFocusStyle, addHoverStyle } from "./Button";

/**
 * Returns a component with all button selector styles applied.
 *
 * @param {ReactElement} component The original component.
 *
 * @returns {ReactElement} Component with applied styles.
 */
const addButtonStyles = flow( [
	/*
	 * Styled-components applies the generated CSS classes in a reversed order,
	 * but we want them in the order: base - hover - focus - active.
	 */
	addActiveStyle,
	addFocusStyle,
	addHoverStyle,
] );

/**
 * Returns a basic styled button.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} Styled button.
 */
const IconLabelledBaseButton = addButtonStyles(
	styled.button`
		display: inline-flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
		cursor: pointer;
		box-sizing: border-box;
		border: 1px solid transparent;
		margin: 0;
		padding: 8px;
		overflow: visible;
		font-family: inherit;
		font-weight: inherit;
		color: ${ props => props.textColor };
		background: ${ props => props.backgroundColor };
		font-size: ${ props => props.textFontSize };

		svg {
			margin: 0 0 8px;
			flex-shrink: 0;
			fill: currentColor;
			// Safari 10
			align-self: center;
		}

		&:active {
			box-shadow: none;
		}
	`
);

/**
 * Returns an icon button that can optionally contain text.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} Styled icon button.
 */
const IconLabeledButton = ( props ) => {
	const { children, icon, textColor } = props;

	const newProps = omit( props, "icon" );

	return (
		<IconLabelledBaseButton { ...newProps }>
			<SvgIcon icon={ icon } color={ textColor } />
			{ children }
		</IconLabelledBaseButton>
	);
};

IconLabeledButton.propTypes = {
	type: PropTypes.string,
	icon: PropTypes.string.isRequired,
	textColor: PropTypes.string,
	textFontSize: PropTypes.string,
	backgroundColor: PropTypes.string,
	borderColor: PropTypes.string,
	hoverColor: PropTypes.string,
	hoverBackgroundColor: PropTypes.string,
	hoverBorderColor: PropTypes.string,
	activeColor: PropTypes.string,
	activeBackgroundColor: PropTypes.string,
	activeBorderColor: PropTypes.string,
	focusColor: PropTypes.string,
	focusBackgroundColor: PropTypes.string,
	focusBorderColor: PropTypes.string,
	focusBoxShadowColor: PropTypes.string,
	children: PropTypes.oneOfType( [
		PropTypes.arrayOf( PropTypes.node ),
		PropTypes.node,
		PropTypes.string,
	] ).isRequired,
};

IconLabeledButton.defaultProps = {
	type: "button",
	textColor: colors.$color_blue,
	textFontSize: "inherit",
	backgroundColor: "transparent",
	borderColor: "transparent",
	hoverColor: colors.$color_white,
	hoverBackgroundColor: colors.$color_blue,
	hoverBorderColor: colors.$color_button_border_hover,
	activeColor: colors.$color_white,
	activeBackgroundColor: colors.$color_blue,
	activeBorderColor: colors.$color_button_border_active,
	focusColor: colors.$color_white,
	focusBackgroundColor: colors.$color_blue,
	focusBorderColor: colors.$color_blue,
	focusBoxShadowColor: colors.$color_blue_dark,
};

export default IconLabeledButton;
