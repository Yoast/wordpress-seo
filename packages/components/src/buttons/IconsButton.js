// External dependencies.
import React from "react";
import PropTypes from "prop-types";

// Internal dependencies.
import { default as Button } from "./Button";
import { SvgIcon } from "../index";

/**
 * Returns an icons button that can optionally contain a prefix and / or a suffix icon.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} styled icon button.
 */
const IconsButton = ( props ) => {
	const {
		children,
		className,
		prefixIcon,
		suffixIcon,
		...buttonProps
	} = props;

	return (
		<Button className={ className } { ...buttonProps }>
			{ prefixIcon && prefixIcon.icon &&
			<SvgIcon
				icon={ prefixIcon.icon }
				color={ prefixIcon.color }
				size={ prefixIcon.size }
			/>
			}
			{ children }
			{ suffixIcon && suffixIcon.icon &&
			<SvgIcon
				icon={ suffixIcon.icon }
				color={ suffixIcon.color }
				size={ suffixIcon.size }
			/>
			}
		</Button>
	);
};

IconsButton.propTypes = {
	className: PropTypes.string,
	prefixIcon: PropTypes.shape( {
		icon: PropTypes.string,
		color: PropTypes.string,
		size: PropTypes.string,
	} ),
	suffixIcon: PropTypes.shape( {
		icon: PropTypes.string,
		color: PropTypes.string,
		size: PropTypes.string,
	} ),
	children: PropTypes.oneOfType( [
		PropTypes.arrayOf( PropTypes.node ),
		PropTypes.node,
		PropTypes.string,
	] ),
};

export default IconsButton;
