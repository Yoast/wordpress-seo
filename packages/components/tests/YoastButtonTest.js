import React from "react";
import renderer from "react-test-renderer";
import { YoastButton } from "../src/buttons/YoastButton";

test( "the YoastButton matches the snapshot", () => {
	const component = renderer.create(
		<YoastButton backgroundColor="#c00" textColor="#fff">ButtonValue</YoastButton>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the YoastButton executes callback", () => {
	const component = renderer.create(
		<YoastButton
			onClick={
				() => {
					return "clicked";
				}
			}
		>ButtonValue</YoastButton>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();

	tree.props.onClick();

	tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
