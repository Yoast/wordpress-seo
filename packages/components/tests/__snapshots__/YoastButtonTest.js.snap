// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the YoastButton executes callback 1`] = `
.c0 {
  color: #fff;
  background: #64a60a;
  min-width: 152px;
  text-shadow: 0 0 2px #000;
  overflow: visible;
  cursor: pointer;
}

.c0::-moz-focus-inner {
  border-width: 0;
}

.c0 span {
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -webkit-justify-content: inherit;
  -ms-flex-pack: inherit;
  justify-content: inherit;
  width: 100%;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  min-height: 48px;
  margin: 0;
  padding: 0 16px;
  padding: 0px 16px;
  border: 0;
  border-radius: 4px;
  box-sizing: border-box;
  font: 400 14px/24px "Open Sans",sans-serif;
  text-transform: uppercase;
  box-shadow: 0 2px 8px 0 rgba( 0,0,0,0.3 );
  -webkit-transition: box-shadow 150ms ease-out;
  transition: box-shadow 150ms ease-out;
}

.c1:hover,
.c1:focus,
.c1:active {
  box-shadow: 0 4px 10px 0 rgba( 0,0,0,0.2 ),inset 0 0 0 100px rgba( 0,0,0,0.1 );
  color: #fff;
}

.c1:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c1::after {
    display: inline-block;
    content: "";
    min-height: 48px;
  }
}

<button
  className="c0 c1"
  onClick={[Function]}
  type="button"
>
  <span>
    ButtonValue
  </span>
</button>
`;

exports[`the YoastButton executes callback 2`] = `
.c0 {
  color: #fff;
  background: #64a60a;
  min-width: 152px;
  text-shadow: 0 0 2px #000;
  overflow: visible;
  cursor: pointer;
}

.c0::-moz-focus-inner {
  border-width: 0;
}

.c0 span {
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -webkit-justify-content: inherit;
  -ms-flex-pack: inherit;
  justify-content: inherit;
  width: 100%;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  min-height: 48px;
  margin: 0;
  padding: 0 16px;
  padding: 0px 16px;
  border: 0;
  border-radius: 4px;
  box-sizing: border-box;
  font: 400 14px/24px "Open Sans",sans-serif;
  text-transform: uppercase;
  box-shadow: 0 2px 8px 0 rgba( 0,0,0,0.3 );
  -webkit-transition: box-shadow 150ms ease-out;
  transition: box-shadow 150ms ease-out;
}

.c1:hover,
.c1:focus,
.c1:active {
  box-shadow: 0 4px 10px 0 rgba( 0,0,0,0.2 ),inset 0 0 0 100px rgba( 0,0,0,0.1 );
  color: #fff;
}

.c1:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c1::after {
    display: inline-block;
    content: "";
    min-height: 48px;
  }
}

<button
  className="c0 c1"
  onClick={[Function]}
  type="button"
>
  <span>
    ButtonValue
  </span>
</button>
`;

exports[`the YoastButton matches the snapshot 1`] = `
.c0 {
  color: #fff;
  background: #c00;
  min-width: 152px;
  text-shadow: 0 0 2px #000;
  overflow: visible;
  cursor: pointer;
}

.c0::-moz-focus-inner {
  border-width: 0;
}

.c0 span {
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -webkit-justify-content: inherit;
  -ms-flex-pack: inherit;
  justify-content: inherit;
  width: 100%;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
  min-height: 48px;
  margin: 0;
  padding: 0 16px;
  padding: 0px 16px;
  border: 0;
  border-radius: 4px;
  box-sizing: border-box;
  font: 400 14px/24px "Open Sans",sans-serif;
  text-transform: uppercase;
  box-shadow: 0 2px 8px 0 rgba( 0,0,0,0.3 );
  -webkit-transition: box-shadow 150ms ease-out;
  transition: box-shadow 150ms ease-out;
}

.c1:hover,
.c1:focus,
.c1:active {
  box-shadow: 0 4px 10px 0 rgba( 0,0,0,0.2 ),inset 0 0 0 100px rgba( 0,0,0,0.1 );
  color: #fff;
}

.c1:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c1::after {
    display: inline-block;
    content: "";
    min-height: 48px;
  }
}

<button
  className="c0 c1"
  type="button"
>
  <span>
    ButtonValue
  </span>
</button>
`;
